### PR TITLE
Teams cubit

### LINKED ISSUE #53: Our Team Screen -> Cubit
### Is your feature request related to a problem? Please describe.

We need to design the cubit for Our Team Screen.

### Describe the solution you'd like

1. should have states: loading, success, error.
2. fake api repository.
3. tests.

### PR DESCRIPTION
**Pull request by TreavVasu**
### Description

Introduces feature #53

**Type of Change:** (Delete irrelevant options)
- Code
- Testing

**Code/Quality Assurance Only** (Delete irrelevant options)

- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

Screenshot_1614448494

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [ ] I have added one/more required dependencies
- [x] I have formatted the code using dart format (dartfmt)

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

**Mention the dependencies/packages/plugins that have been added (if any)**

### CODE REVIEW

**Comment by TreavVasu** [comment]
API response is the one that server gives now.

File: lib/main.dart [hunk 1]
```diff
@@ -58,7 +61,9 @@ class ECellApp extends StatelessWidget {
               create: (_) => EventsCubit(FakeEventsRepository()), child: EventsScreen()),
           S.routeSponsors: (_) => BlocProvider(
               create: (_) => SponsorsCubit(APISponsorsRepository()), child: SponsorsScreen()),
-          S.routeEsummit: (_) => ESummitScreen()
+          S.routeEsummit: (_) => ESummitScreen(),
+          S.routeTeam: (_) =>
+              BlocProvider(create: (_) => TeamCubit(FakeTeamRepository()), child: TeamScreen()),
```
Comment (werainkhatri) [inline lib/main.dart]: TeamScreen is supposed to be a tab in about us section. so there should be a routeAboutUs routing us to AboutUsScreen().

**Review by werainkhatri — CHANGES_REQUESTED** [review]
1. Move the teams screen to the about_us screen as a tab
2. merge TreavVasu#6

File: lib/main.dart [hunk 2]
```diff
@@ -58,11 +58,11 @@ class ECellApp extends StatelessWidget {
           S.routeSpeaker: (_) => BlocProvider(
               create: (_) => SpeakerCubit(APISpeakerRepository()), child: SpeakerScreen()),
           S.routeEvents: (_) => BlocProvider(
-              create: (_) => EventsCubit(FakeEventsRepository()), child: EventsScreen()),
+              create: (_) => EventsCubit(APIEventsRepository()), child: EventsScreen()),
```
Comment (werainkhatri) [inline lib/main.dart]: that was fake for a reason.

File: lib/main.dart (hunk 1 above)
Comment (werainkhatri) [reply lib/main.dart]: This is ondone.

**Review by werainkhatri — CHANGES_REQUESTED** [review]

File: lib/main.dart [hunk 3]
```diff
@@ -58,11 +58,11 @@ class ECellApp extends StatelessWidget {
...
               create: (_) => SpeakerCubit(APISpeakerRepository()), child: SpeakerScreen()),
           S.routeEvents: (_) => BlocProvider(
-              create: (_) => EventsCubit(APIEventsRepository()), child: EventsScreen()),
+              create: (_) => EventsCubit(FakeEventsRepository()), child: EventsScreen()),
           S.routeSponsors: (_) => BlocProvider(
               create: (_) => SponsorsCubit(APISponsorsRepository()), child: SponsorsScreen()),
           S.routeEsummit: (_) => ESummitScreen(),
-          S.routeAboutUs: (_) =>
+          S.routeTeam: (_) =>
               BlocProvider(create: (_) => TeamCubit(FakeTeamRepository()), child: TeamScreen()),
```
Comment (werainkhatri) [inline lib/main.dart]:
```suggestion
          S.routeAboutUs: (_) =>
              BlocProvider(create: (_) => TeamCubit(FakeTeamRepository()), child: AboutUsScreen()),
```

**Review by werainkhatri — CHANGES_REQUESTED** [review]

**Review by werainkhatri — APPROVED** [review]

File: lib/main.dart [hunk 4]
```diff
@@ -7,6 +7,10 @@ import 'core/res/strings.dart';
 import 'core/themes/app_theme.dart';
 import 'core/utils/injection.dart';
 import 'models/global_state.dart';
+import 'screens/about_us/about_us.dart';
+import 'screens/about_us/tabs/team/cubit/team_cubit.dart';
+import 'screens/about_us/tabs/team/team.dart';
```
Comment (werainkhatri) [inline lib/main.dart:12]:
```suggestion
```

**Review by werainkhatri — APPROVED** [review]

**Review by TreavVasu — COMMENTED** [review]
Done

**Pull request merged**